src/main/java/gov/usgs/traveltime/TTService.java

```diff
@@ -0,0 +1,274 @@
+      ttLocal.newSession(
+          request.EarthModel,
+          request.Source.Depth,
+          phases,
+          request.Source.Latitude,
+          request.Source.Longitude,
+          request.ReturnAllPhases,
+          request.ReturnBackBranches,
+          request.ConvertTectonic,
+          false);
+
+      // for each Receiver in the request
+      responseList = new ArrayList<TravelTimeReceiver>();
+      for (TravelTimeReceiver receiver : request.Receivers) {
+
+        // get the traveltime for this source/receiver pair
+        // based on what we got
+        double lat = Double.NaN;
+        if (receiver.Latitude != null) {
+          lat = receiver.Latitude;
+        }
+
+        double lon = Double.NaN;
+        if (receiver.Longitude != null) {
+          lon = receiver.Longitude;
+        }
+
+        double elev = Double.NaN;
+        if (receiver.Elevation != null) {
+          elev = receiver.Elevation;
+        }
+
+        double delta = Double.NaN;
+        if (receiver.Distance != null) {
+          delta = receiver.Distance;
+        }
+
+        double azimuth = Double.NaN;
+
+        TTime travelTime = ttLocal.getTT(lat, lon, elev, delta, azimuth);
+
+        if (travelTime == null) {
+          continue;
+        }
+
+        // create response
+        TravelTimeReceiver response = new TravelTimeReceiver(receiver);
+        response.Branches = new ArrayList<TravelTimeData>();
+
+        // add traveltimes to response
+        for (int i = 0; i < travelTime.getNumPhases(); i++) {
+          TTimeData phase = travelTime.getPhase(i);
+
+          // build travel time data
+          TravelTimeData data = new TravelTimeData();
+          data.Phase = phase.phCode;
+          data.TravelTime = phase.tt;
+          data.DistanceDerivative = phase.dTdD;
+          data.DepthDerivative = phase.dTdZ;
+          data.RayDerivative = phase.dXdP;
+          data.StatisticalSpread = phase.spread;
+          data.Observability = phase.observ;
+          data.TeleseismicPhaseGroup = phase.phGroup;
+          data.AuxiliaryPhaseGroup = phase.auxGroup;
+          data.LocationUseFlag = phase.canUse;
+          data.AssociationWeightFlag = phase.dis;
+
+          // add to list
+          response.Branches.add(data);
+        }
+
+        // add response to response list
+        responseList.add(response);
+      }
+    } catch (BadDepthException e) {
+      throw new TravelTimeException("Source depth out of range");
+    } catch (TauIntegralException e) {
+      throw new TravelTimeException("Bad Tau Integral");
+    } catch (IOException e) {
+      throw new TravelTimeException("Read Error");
+    } catch (ClassNotFoundException e) {
+      throw new TravelTimeException("Input serialization");
+    }
+
+    request.Response = responseList;
+
+    return (request);
+  }
+
+  /**
+   * Function to get a travel time using the provided input, implementing the TravelTime service
```

Comment:
I think this is a cut and paste, the getTravelTimePlot function has the exact same description as the getTravelTimes function...where this is getting the travel time model (travel times for phases over all distances from source providing the full travel time curve (aka plot) for each phase)...